homeassistant/components/wemo/__init__.py

```diff
@@ -163,7 +174,47 @@ def stop_wemo(event):
                 device,
             )
 
-    return True
+        self._added_serial_numbers.add(device.serialnumber)
+
+
+class WemoDiscovery:
+    """Use SSDP to discover WeMo devices."""
+
+    ADDITIONAL_SECONDS_BETWEEN_SCANS = 10
+    MAX_SECONDS_BETWEEN_SCANS = 300
+
+    def __init__(self, hass: HomeAssistant, wemo_dispatcher: WemoDispatcher) -> None:
+        """Initialize the WemoDiscovery."""
+        self._hass = hass
+        self._wemo_dispatcher = wemo_dispatcher
+        self._stop = None
+        self._scan_delay = 0
+
+    async def async_discover_and_schedule(self, *_) -> None:
+        """Periodically scan the network looking for WeMo devices."""
+        _LOGGER.debug("Scanning network for WeMo devices...")
+        try:
+            for device in await self._hass.async_add_executor_job(
+                pywemo.discover_devices
+            ):
+                self._wemo_dispatcher.async_add_unique_device(self._hass, device)
+        finally:
+            # Run discovery more frequently after hass has just started.
+            self._scan_delay = min(
+                self._scan_delay + self.ADDITIONAL_SECONDS_BETWEEN_SCANS,
+                self.MAX_SECONDS_BETWEEN_SCANS,
+            )
+            self._stop = async_call_later(
+                self._hass,
+                self._scan_delay,
+                self.async_discover_and_schedule,
+            )
+
+    @callback
+    def async_stop_discovery(self) -> None:
+        """Stop the periodic background scanning."""
+        if self._stop:
+            self._stop()
 
```

Comment:
Probably set `self._stop` to None after calling it.

Reply:
Done